client/data/account/actions.js

```diff
@@ -0,0 +1,37 @@
+import { dispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { apiFetch } from '@wordpress/data-controls';
+import { NAMESPACE } from '../constants';
+import ACTION_TYPES from './action-types';
+
+export function updateAccount( payload ) {
+	return {
+		type: ACTION_TYPES.SET_ACCOUNT,
+		payload,
+	};
+}
+
+export function updateIsRefreshingAccount( isRefreshing ) {
+	return {
+		type: ACTION_TYPES.SET_IS_REFRESHING,
+		isRefreshing,
+	};
+}
+
+export function* refreshAccount() {
```

Comment:
This may be something standard but I am new to this. So trying to understand better.
Why is this `refreshAccount` in the `actions` file instead of `resolvers`?

Reply:
All `resolvers` also have a matching function in `selectors`.
When a selector has never been called before, WP data calls the matching resolver.
Once the resolver has returned some data, the matching selector is run.
This is useful when there is no data in the redux store for a selector so that the matching resolver can fetch data instead.

In this case, `refreshAccount` is an action that we're dispatching every time we need to refresh the account data (but keeping the old one in the redux store).

I tried explaining the responsibilities of a resolver with my own words, but I think the docs do a better job :D https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#resolvers

Let me know if this helps answer your question or if I misunderstood it! :)

Reply:
Thanks a bunch. Your explanation is to the point and I got my answer :100: